app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,72 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all.order(:created_at)
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      flash[:error] = "Could not find task with id: #{task_id}"
+      redirect_to task_path
```

Comment:
Actually, when you manually test this, _because_ it redirects back to the `task_path`, which... is this action... and then it gets into this case again... which redirects back to the `task_path`... it gets into an infinite loop. You'll probably want to fix this by redirecting somewhere else